temporian/core/evaluator.py

```diff
@@ -52,72 +52,76 @@ def evaluate(
             f" {type(query)}."
         )
 
-    # get features from events
-    features_to_compute = [
-        feature
-        for event in events_to_compute
-        for feature in event.features()  # pytype: disable=attribute-error
-    ]
-
     # get backend
     selected_backend = backends.BACKENDS[backend]
     event = selected_backend["event"]
     evaluate_schedule_fn = selected_backend["evaluate_schedule_fn"]
     read_csv_fn = selected_backend["read_csv_fn"]
 
-    # calculate opeartor schedule
-    schedule = get_operator_schedule(features_to_compute)
+    # get features from events
+    features_to_compute = {
+        feature
+        for event in events_to_compute
+        for feature in event.features()  # pytype: disable=attribute-error
+    }
+    # calculate opeartor schedule. Only using keys (operators) for now, discarding
+    # depth
+    schedule = get_operator_schedule(features_to_compute).keys()
 
     # materialize input data. TODO: separate this logic
-    materialized_input_data = {}
-    for input_event, input_event_spec in input_data.items():
-        if not isinstance(input_event_spec, event):
-            input_event_spec = read_csv_fn(
-                input_event_spec, input_event.sampling()
-            )
-        materialized_input_data[input_event] = input_event_spec
-
+    materialized_input_data = {
+        input_event: (
+            input_event_spec
+            if isinstance(input_event_spec, event)
+            else read_csv_fn(input_event_spec, input_event.sampling())
```

Comment:
For later / add todo: Add a "isinstance" check in the csv case, and raise an exception if input_event_spec's type is not supported.